templates/ingress.yaml

```diff
@@ -372,88 +372,73 @@ metadata:
   {{- end }}
 spec:
 {{- include "coder.ingress.tls" . }}
-  backend:
-    serviceName: {{ include "coder.serviceName" . }}
-    servicePort: 8080
+  defaultBackend:
+    service:
+      name: {{ include "coder.serviceName" . }}
+      port:
+        number: 8080
   {{- if not .Values.coderd.replica.enable }}
   rules:
   - host: {{ .Values.ingress.host | quote }}
     http:
       paths:
-    {{- if or (not .Values.ingress.usePathWildcards) .Values.ingress.useDefault }}
+    {{- if .Values.ingress.useDefault }}
       {{- if hasKey .Values "envproxy" }}
-      - path: /proxy/
+      - path: /proxy
+        pathType: Prefix
         backend:
-          serviceName: envproxy
-          servicePort: 8080
+          service:
+            name: envproxy
+            port:
+              number: 8080
       {{- end }}
-      - path: /api/
+      - path: /api
+        pathType: Prefix
         backend:
-          serviceName: {{ include "coder.serviceName" . }}
-          servicePort: 8080
-      - path: /auth/
+          service:
+            name: {{ include "coder.serviceName" . }}
+            port:
+              number: 8080
+      - path: /auth
+        pathType: Prefix
         backend:
-          serviceName: {{ include "coder.serviceName" . }}
-          servicePort: 8080
+          service:
+            name: {{ include "coder.serviceName" . }}
+            port:
+              number: 8080
       - path: /
+        pathType: Prefix
         backend:
           {{- if hasKey .Values "dashboard" }}
-          serviceName: dashboard
-          servicePort: 3000
+          service:
+            name: dashboard
+            port:
+              number: 3000
           {{- else }}
-          serviceName: {{ include "coder.serviceName" . }}
-          servicePort: 8080
+          service:
+            name: {{ include "coder.serviceName" . }}
+            port:
+              number: 8080
           {{- end }}
-    {{- if ne .Values.devurls.host "" }}
+      {{- if ne .Values.devurls.host "" }}
   - host: {{ .Values.devurls.host | quote }}
     http:
       paths:
       - path: /
+        pathType: Prefix
         backend:
           {{- if hasKey .Values "envproxy" }}
-          serviceName: envproxy
-          servicePort: 8080
+          service:
+            name: envproxy
+            port:
+              number: 8080
           {{- else }}
-          serviceName: {{ include "coder.serviceName" . }}
-          servicePort: 8080
+          service:
+            name: {{ include "coder.serviceName" . }}
+            port:
+              number: 8080
           {{- end }}
-    {{- end }}
-    {{- else }}
-      - path: /proxy/*
```

Comment:
@coadler Removed all this stuff, `pathType: Prefix` should cover these, I think. I tested with my dev deploy but that's it so far.